active/ItemsSourceView-Indexof/ItemsSourceView-IndexOf.md

```diff
@@ -0,0 +1,61 @@
+# Background
+<!-- Use this section to provide background context for the new API(s) 
+in this spec. -->
+The Xaml [ItemsRepeater](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsRepeater) control displays a list of objects using a specified template (similar to a [ListView](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ListView)). You provide the list of objects with the [ItemsRepeater.ItemsSource](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsRepeater.ItemsSource) property. The list need only be an IIterable but can be an IVector for better perf. ItemsRepeater exposes a read-only [ItemsRepeater.ItemsSourceView](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsRepeater.ItemsSourceView) property that provides a view on this list (an [ItemsSourceView](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsSourceView)), which has helpers like Count and GetAt over what might be just an IIterable.
+
+Currently the ItemsSourceView class does not expose an IndexOf function. However during development, this method was used multiple times internally on derivatives of the class. To make the usage of this existing function easier on subclasses, this spec adds an IndexOf function to the ItemsSourceView.
+
+# Description
+
+The IndexOf function returns the index of the given element in the items source the ItemsSourceView is used for.
+Note that if an element is not contained in the items source, the function will return -1.
+> Spec note: returning -1 matches the behavior of for example IndexOf on [UIElementCollection](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.UIElementCollection) and List<T>.
+# Examples
+C# example
+
+```c#
+/// ************** Behavior ************ ///
+
+var items = new List<string>() { "One", "Two" };
+var itemsSourceView = new ItemsSourceView(items);
+var secondItem = itemsSourceView.GetAt(1);
+Assert.AreEqual(1, itemsSourceView.IndexOf(secondItem);
+
+
+/// ****** Use case in event handler ****** ///
+
+public void ItemInvoked(object sender, RoutedEventArgs e)
```

Comment:
What event is this listening to? Is the sender really an item, rather than an element?

Reply:
I think in the case of a collection of controls, instead of wrapping them, we render them directly, so there is a scenario where the sender could be the item.

Reply:
For the example we should be specific about what event it's listening to. We want the example code to be highly copy/pasteable. What was the code you were writing when you originally needed this?

Reply:
There wasn't a specific reason. Since the context would be a bit large for a sample, we can also get rid of it. What do you think?

Reply:
I think the event handler code is a bit confusing. Perhaps removing that part would be better. We could also make it more concrete and show the event handler for one of ItemsRepeater's events like ElementPrepared or ElementIndexChanged if we want to show usage in a handler.

Reply:
I think it's OK to remove it.

Reply:
I've removed it now.